docs/chapters/03-2.md

```diff
@@ -0,0 +1,174 @@
+
+The first-layer, simple convolution/pool combinations usually detect simple features, such as oriented edge detections. After the first convolution/pool layer, the objective is to detect combinations of features from previous layers. To do this, steps 2 to 4 are repeated with multiple kernels over previous-layer feature maps, and are summed in a new feature map:
+
+1. A new 5x5 kernel is slided over all feature maps from pervious layers, with results summed up. (Note: In Prof. LeCun's experiment in 1989 the connection is not full for computation purpose. Modern settings usually enforce full connections): size *10x10*
+2. Pass the output of the convolution to a nonlinear function: size *10x10*
+3. Repeat 1/2 for 16 kernels. 
+4. Pass the result to the pooling layer that averages over 2x2 window with stride 2: size *5x5* each feature map
+
+To generate an output, the last layer of convolution is conducted, which seems like a full connections but indeed is convolutional.
+
+1. The final convolution layer slides a 5x5 kernel over all feature maps, with results summed up: size *1x1*
+2. Pass through nonlinear function: size *1x1*
+3. Generate the single output for one category.
+4. Repeat all pervious steps for each of the 10 categories(in parallel)
+
+See [this animation](http://cs231n.github.io/convolutional-networks/) on Andrej Karpathy's website on how convolutions change the shape of the next layer's feature maps. Full paper can be found [here](https://papers.nips.cc/paper/293-handwritten-digit-recognition-with-a-back-propagation-network.pdf).
+
+### [Shift Equivariance](1:05:52-1:09:25)
+<center>
+    <img src="shift_invariance.gif" width="300px" /><br>
+    <b>Fig. 2</b> Shift Equivariance
+</center>
+
+As demonstrated by the animation on the slides(here's another example), translating the input image results in same translation of the feature maps. However, the changes in feature maps are scaled by convolution/pooling operations. E.g. the 2x2 pooling with stride 2 will reduce the 1-pixel shift in input layer to 0.5-pixel shift in the following feature maps. Spatial resolution is then exchanged for incraesed number of feature types, i.e. making the representation more abstract and less sensitive to shifts and distortions.
+
+### [Overall Architecture Breakdown](1:09:25-1:11:20)
+
+Generic CNN architecture can broken down into several basic layer archetypes:
+
+* Normalization layers - adjusting whitening (optional)
+  * Subtratrive methods e.g. average removal, high pass filtering
+  * Divisive: local contrast normalization, variance normalization
+* Filter Banks: increasing dimensionality, finding edges, etc.
+* Non-linearities: sparsification
+  * Typically Rectified Linear Unit (ReLU): $\text{ReLU}(x) = \text{max}(x, 0)$
+* Pooling: aggregating over a feature map:
+  * $\text{MAX}= \text{Max}_i(X_i)$
+  * $\text{L}p= \left(\sum_{i=1}^n \|X_i\|^p \right)^{\frac{1}{p}}$
+  * $\text{Prob}= \frac{1}{b} \left(\sum_{i=1}^n e^{b X_i} \right)$
```

Comment:
Ugly!

Reply:
Alf, what is ugly? Do you mean you prefer \texttt{MAX} instead of \text{MAX} or something?